apps/agreement/contracts/Agreement.sol

```diff
@@ -0,0 +1,1350 @@
+        uint256 challengeAmount;            // Amount of collateral token that will be locked from the challenger's own balance every time an action is challenged
+        uint64 challengeDuration;           // Challenge duration in seconds, during which the submitter can raise a dispute
+        Staking staking;                    // Staking pool cache for the collateral token
+    }
+
+    struct DisputableInfo {
+        bool registered;                                                    // Whether a Disputable app is registered
+        uint256 collateralRequirementsLength;                               // Identification number of the next collateral requirement instance
+        mapping (uint256 => CollateralRequirement) collateralRequirements;  // List of collateral requirements indexed by id
+    }
+
+    StakingFactory public stakingFactory;                           // Staking factory to be used for the collateral staking pools
+
+    uint256 private settingsLength;
+    mapping (uint256 => Setting) private settings;                  // List of historic settings indexed by ID
+    mapping (address => uint256) private lastSettingSignedBy;       // List of last setting signed by user
+
+    uint256 private actionsLength;
+    mapping (uint256 => Action) private actions;                    // List of actions indexed by ID
+    mapping (address => DisputableInfo) private disputableInfos;    // Mapping of disputable address => disputable infos
+
+    uint256 private challengesLength;
+    mapping (uint256 => Challenge) private challenges;              // List of challenges indexed by ID
+    mapping (uint256 => uint256) private challengeByDispute;        // Mapping of arbitrator's dispute ID => challenge ID
+
+    /**
+    * @notice Initialize Agreement app for "`_title`" and content "`_content`", with arbitrator `_arbitrator` and staking factory `_factory`
+    * @param _title String indicating a short description
+    * @param _content Link to a human-readable text that describes the initial rules for the Agreements instance
+    * @param _arbitrator Address of the IArbitrator that will be used to resolve disputes
+    * @param _stakingFactory Staking factory to be used for the collateral staking pools
+    */
+    function initialize(string _title, bytes _content, IArbitrator _arbitrator, StakingFactory _stakingFactory) external {
+        initialized();
+        require(isContract(address(_stakingFactory)), ERROR_STAKING_FACTORY_NOT_CONTRACT);
+
+        stakingFactory = _stakingFactory;
+
+        settingsLength++; // Setting ID zero is considered the null setting for further validations
+        _newSetting(_title, _content, _arbitrator);
+    }
+
+    /**
+    * @notice Register `_disputable`, setting its collateral requirements to:
+    * @notice - `@tokenAmount(_collateralToken: address, _actionAmount)` for submissions
+    * @notice - `@tokenAmount(_collateralToken: address, _challengeAmount)` for challenges
```

Comment:
It's unlikely we'll be able to show this via radspec at the moment, so I wonder if we shouldn't just remove them.